src/app/shared/components/button/button.component.ts

```diff
@@ -1,16 +1,28 @@
-import { Component } from '@angular/core';
+import { Component, Input } from '@angular/core';
 import { FaIconLibrary } from '@fortawesome/angular-fontawesome';
+import { IconName, IconProp } from '@fortawesome/fontawesome-svg-core';
 import { fas } from '@fortawesome/free-solid-svg-icons';
 
 @Component({
   selector: 'app-button',
   templateUrl: './button.component.html',
   styleUrls: ['./button.component.scss']
 })
-export class ButtonComponent {
+export class ButtonComponent{
+  @Input() buttonText?:string;
+  @Input() iconName?:'trash-can' | 'pencil';
```

Comment:
I see that the intent was to declare that an iconName can have only two values. Either a trash-can or a pencil. But how about the further requirements, let's just say we want to include more and more icons later? Probably an enumeration is more appropriate and clean to do the job. You can declare it and use it across the app. Basically it will eliminate typo type issues.

Reply:
Made an enum for it